src/charm.py

```diff
@@ -218,6 +238,24 @@ def server_cert_path(self) -> Optional[str]:
     # === UTILITY METHODS === #
     ###########################
 
+    def _update_mimir_cluster(self):  # common exit hook
+        # TODO: does this prevent disabling log forwarding on relation-broken ???
+        # if not self.coordinator.is_coherent():
+        #     return
+        tls = self._is_cert_available
+
+        s3_config_data = self._get_s3_storage_config()
+
+        self.cluster_provider.publish_data(
```

Comment:
I'd add a couple of comments here or a docstring to explain the holistic logic going on here. Be kind to your future self